lib/utils/types.ts

```diff
@@ -14,22 +14,15 @@ export type Tagged<T, S, A> = t.Type<S, A & T>;
 export const tag = <T>() => <S, A>(type: t.Type<S, A>): Tagged<T, S, A> =>
   type as any;
 
-const getObjectValues = <T extends object>(obj: T): ReadonlyArray<string> =>
-  Object.keys(obj).reduce<ReadonlyArray<string>>(
-    (acc, key) => [...acc, (obj as any)[key]],
-    []
-  );
-
 /**
  * Creates an io-ts Type from an enum
  */
 export const enumType = <E>(e: {}, name: string): t.Type<any, E> => {
-  const values = getObjectValues(e);
```

Comment:
why not just check for e[v] ?

Reply:
The [Object.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) method returns an array of a given object's own enumerable properties, skipping inherited properties. 

For instance `__proto__` (an inherited property) is a valid property for `e` but not a valid enum value.

Reply:
I've added a check with hasOwnProperties()

Reply:
Sorry only now I realized that you're validating the value and not the key, I'll revert it back :)